### ISSUE TITLE
GeoAxis background color

### ISSUE THREAD

**Issue by Ethan-Russell** [MakieOrg/GeoMakie.jl#193]
With the new and improved GeoAxis, I see that there is no option to add a background color to the axis.  Is it simple to add that feature?

**Comment by SimonDanisch** [comment]
You could add it like in the normal Makie axis: https://github.com/MakieOrg/Makie.jl/blob/master/src/makielayout/blocks/axis.jl#L194-L196
Somewhere here I suppose: 
https://github.com/MakieOrg/GeoMakie.jl/blob/master/src/geoaxis.jl#L710

**Comment by jpinelo** [comment]
# Polygon and surface wrapping artefacts at projection boundary (Robinson)

## Summary

Land polygons and `surface!` meshes that span wide longitude ranges at high latitudes produce visible artefacts on non-cylindrical projections (Robinson, Mollweide, etc.). Edges that should follow the curved projection boundary instead cut straight across the map in screen space. This makes GeoMakie unsuitable for publication-quality global base maps on these projections without significant workarounds.

As an exemple, Cartopy (Python) handles this transparently via automatic polygon clipping at the projection boundary. GeoMakie currently has no equivalent.

## Minimal reproducer

```julia
using CairoMakie, GeoMakie

fig = Figure(size=(800, 500))
ga = GeoAxis(fig[1, 1]; dest="+proj=robin +lon_0=150")

# Land polygons — Russia/Canada/Antarctica wrap across the top and bottom
land = GeoMakie.land()
poly!(ga, land; color=:lightgrey, strokecolor=:grey, strokewidth=0.4)

fig
# save("geomakie_wrapping.png", fig)
```

Expected: filled land polygons clipped to the Robinson ellipse boundary, as in Cartopy.

Actual: horizontal bands/lines at the top and bottom of the projection where polygon edges from high-latitude land masses (Russia, Arctic Canada, Antarctica) connect across the projection boundary rather than following it.

## What Cartopy does

In Python's Cartopy, `ax.add_feature(cfeature.LAND)` automatically clips every polygon to the projection boundary before rendering. The user never sees wrapping artefacts. The relevant Cartopy machinery splits polygons that cross the projection's antimeridian or boundary into multiple sub-polygons, each of which renders correctly within the projected domain.

## Workarounds attempted

We tried several approaches during a port of a two-panel Robinson projection figure from Python/Cartopy to Julia/GeoMakie:

1. **`poly!` with `GeoMakie.land()`** — artefacts from Antarctica (extends to -90°) and wide-longitude Arctic land masses wrapping across the projection boundary.

2. **`poly!` with `naturalearth("land", 110)` + latitude filtering** — attempted to skip polygons with vertices beyond ±78° latitude. Filter did not reliably exclude problematic polygons (likely due to MultiPolygon nesting in the GeoInterface coordinate extraction).

3. **`lines!` with `GeoMakie.coastlines()`** — same wrapping artefact, now as lines instead of filled polygons.

4. **`surface!` with uniform colour field** — used for ocean background fill since `GeoAxis` has no `backgroundcolor` (see #193). The surface mesh triangles also wrap at the antimeridian, creating a visible band across the map. Shifting the longitude array to align with the projection's antimeridian (`circshift` approach from the docs) did not eliminate the artefact.

5. **`surface!` with raster land-ocean mask** — pre-computed a 0.5° point-in-polygon mask to avoid vector polygons entirely. The surface mesh still wraps at the antimeridian.

6. **`ylims!` to constrain visible area** — does not clip rendered content, only changes axis limits.

7. **`hidedecorations!`** — confirmed artefacts are not from axis spines or gridlines.

None of these produced a clean Robinson projection base map comparable to Cartopy's output.

## What would fix this

The core missing capability is **automatic polygon/geometry clipping at the projection boundary**, equivalent to what Cartopy does internally. Specifically:

- **For `poly!` on `GeoAxis`:** polygons that cross the projection boundary should be split into sub-polygons that each render within the projected domain.
- **For `surface!` on `GeoAxis`:** mesh triangles that span the antimeridian should be handled (either by splitting the mesh or by discarding/clipping problematic triangles).
- **`backgroundcolor` for `GeoAxis`** (already tracked in #193) would eliminate the need for the `surface!` ocean-fill workaround entirely.

## Context

This came up while porting a scientific figure (two-panel Robinson projection comparing ISW literature coverage with operational satellite detections) from Python/Cartopy/Matplotlib to Julia/GeoMakie/CairoMakie. The data pipeline ported cleanly; the base map rendering was the blocker. We ultimately kept the Python version for this figure.

## Environment

- GeoMakie v0.7.x (latest at time of testing)
- CairoMakie (latest)
- Julia 1.10.10
- macOS (Apple Silicon)

<img width="764" height="831" alt="Image" src="https://github.com/user-attachments/assets/51e09cb0-dbbd-4fba-b66b-9bf9646e2da5" />

<img width="593" height="661" alt="Image" src="https://github.com/user-attachments/assets/9b75c277-3d59-4c1b-a2b8-efca888675fd" />

<img width="560" height="658" alt="Image" src="https://github.com/user-attachments/assets/7b6c6381-df63-455c-8c5f-9095d7af4497" />

**Comment by asinghvi17** [comment]
Yup, the main three features we'd need here are the following:
- Knowing, in destination space, what the geometry of the projection's boundary is (port-able from Cartopy)
- Being able to perform polygon clipping in spherical space (coming soon in GeometryOps for arbitrary polygons)
- Being able to inject a stage into Makie polygon, line, surface and mesh rendering somehow, that would perform this clipping in the presence of a known transformation

The first two are doable and in progress, but the last one is _very_ tricky.  cc @SimonDanisch since we were discussing this at some point

**Comment by jpinelo** [comment]
Thanks for the clear breakdown. Happy to test against our use case when any of these pieces land — we have the full Julia pipeline ready, just blocked on the base map rendering. Watching the issue.
